libraries/ESP8266WiFi/examples/BearSSL_CertStore/BearSSL_CertStore.ino

```diff
@@ -40,7 +40,83 @@ const char *pass = "....";
 // A single, global CertStore which can be used by all
 // connections.  Needs to stay live the entire time any of
 // the WiFiClientBearSSLs are present.
-CertStoreSPIFFSBearSSL certStore;
+BearSSL::CertStore certStore;
+
+// Uncomment below to use the SD card to store the certs
+// #define USE_SDCARD 1
+
+#ifdef USE_SDCARD
+
+#include <SD.h>
+class SDCertStoreFile : public BearSSL::CertStoreFile {
+  public:
+    SDCertStoreFile(const char *name) {
+      _name = name;
+    };
+    virtual ~SDCertStoreFile() override {};
+
+    // The main API
+    virtual bool open(bool write = false) override {
+      _file = SD.open(_name, write ? FILE_WRITE : FILE_READ);
+      return _file;
+    }
+    virtual bool seek(size_t absolute_pos) override {
+      return _file.seek(absolute_pos);
+    }
+    virtual ssize_t read(void *dest, size_t bytes) override {
+      return _file.read(dest, bytes);
+    }
+    virtual ssize_t write(void *dest, size_t bytes) override {
+      return _file.write((const uint8_t*)dest, bytes);
+    }
+    virtual void close() override {
+      _file.close();
+    }
+
+  private:
+    File _file;
+    const char *_name;
+};
+
+SDCertStoreFile certs_idx("/certs.idx");
+SDCertStoreFile certs_ar("/certs.ar");
+
+#else
+
+#include <FS.h>
+class SPIFFSCertStoreFile : public BearSSL::CertStoreFile {
+  public:
+    SPIFFSCertStoreFile(const char *name) {
+      _name = name;
+    };
+    virtual ~SPIFFSCertStoreFile() override {};
```

Comment:
I'm running against the clock at wörk, so I can't test this, but off the top of my head I came up with [this](https://gist.github.com/devyte/22cc237e50d58a03ea43ce37c425cb14). This can be part of the repo, and worst case the user would just need to instantiate. Or we can instantiate based on a #define.